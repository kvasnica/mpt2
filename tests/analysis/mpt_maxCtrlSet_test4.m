function mpt_maxCtrlSet_test4

% tests the "volume convergence" option:
% http://control.ee.ethz.ch/~mpt/hg/mpt?cs=80e99ef5e36f

lti1d

% standard computation has Options.Vconverge disabled and requires 4 iterations:
[p,q,r]=mpt_maxCtrlSet(sysStruct, struct('Kinf', 1, 'verbose', 0));
mbg_assertequal(q, 4);
mbg_asserttrue(p==unitbox(1,4));  % also test that the Kinf set is correct

% the 4 iterations have volume increase as follows:
% 1. 61.8 %
% 2. 38.2 %
% 3. 10.6 %
% 4. 0 %
%
% therefore we set Options.Vconverge=20 which should cause the algorithm to
% break at iteration 3:
[p,q,r]=mpt_maxCtrlSet(sysStruct, struct('Kinf', 1, 'verbose', 0, 'Vconverge', 20));
mbg_assertequal(q, 3);
