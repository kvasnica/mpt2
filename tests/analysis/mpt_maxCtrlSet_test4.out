Iteration Number 1
Iteration 1       
Iteration Number 1
Iteration 1       
Volume difference (10.56 %) below tolerance (20.00 %), aborting computation...
